### PR TITLE
Fix an automatic rebase against the upstream

### LINKED ISSUE #21826: Che-Code automatic rebase against upstream VS Code is failed
### Is your task related to a problem? Please describe

The GitHub Workflow for Che-Code automatic rebase against upstream VS Code is failed:
https://github.com/che-incubator/che-code/actions/runs/3512719621/jobs/5884724537

```
Run ./rebase.sh
  ./rebase.sh
  shell: /usr/bin/bash -e {0}
Using git /usr/bin/git git version [2](https://github.com/che-incubator/che-code/actions/runs/3512719621/jobs/5884724537#step:7:2).[3](https://github.com/che-incubator/che-code/actions/runs/3512719621/jobs/5884724537#step:7:3)8.1
⚠️  There are conflicting files, trying to solve...
 ➡️  Analyzing conflict for code/extensions/package.json
  ⚙️ reworking code/extensions/package.json...
  ⚙️  applying override settings from .rebase/override/code/extensions/package.json...
 ➡️  Analyzing conflict for code/extensions/yarn.lock
code/extensions/yarn.lock file cannot be automatically rebased. Aborting
Error: Process completed with exit code 1.
```

### Describe the solution you'd like

Fix the incompatibility with the upstream.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by azatsarynnyy**
### What does this PR do?
Fixes an auto-rebase against the upstream.

### What issues does this PR fix?

fixes https://github.com/eclipse/che/issues/21826

### How to test this PR?
All the PR status checks must be green.

### CODE REVIEW

File: .rebase/override/code/extensions/package.json [hunk 1]
```diff
@@ -1,5 +0,0 @@
-{
-  "devDependencies": {
-    "esbuild": "^0.14.49"
```
Comment (azatsarynnyy) [inline .rebase/override/code/extensions/package.json:3]: It was introduced by https://github.com/che-incubator/che-code/pull/86.
Currently, it looks obsolete as the `esbuild` version [was bumped upstream recently](https://github.com/microsoft/vscode/commit/7e62f31610a3a6932ed83464ce48d64561c828ac) to `0.15.14`.
https://github.com/che-incubator/che-code/pull/86 was partially reverted by https://github.com/che-incubator/che-code/pull/137. Here is the 2'nd part.
@RomanNikitenko we don't need that anymore, right?

File: rebase.sh [hunk 2]
```diff
@@ -212,8 +212,6 @@ resolve_conflicts() {
       apply_code_package_changes
     elif [[ "$conflictingFile" == "code/product.json" ]]; then
       apply_code_product_changes
-    elif [[ "$conflictingFile" == "code/extensions/package.json" ]]; then
-      apply_package_changes_by_path "code/extensions/package.json"
```
Comment (azatsarynnyy) [inline rebase.sh:216]: https://github.com/che-incubator/che-code/pull/152#pullrequestreview-1188068551

File: .rebase/override/code/extensions/package.json (hunk 1 above)
Comment (RomanNikitenko) [reply .rebase/override/code/extensions/package.json:3]: yes, taking into account
> the esbuild version [was bumped upstream recently](https://github.com/microsoft/vscode/commit/7e62f31610a3a6932ed83464ce48d64561c828ac) to 0.15.14

we can remove it

File: rebase.sh (hunk 2 above)
Comment (RomanNikitenko) [reply rebase.sh:216]: agree, now we can remove it

**Review by RomanNikitenko — APPROVED** [review]

**Pull request merged**